src/components/settings/SettingsLMSTab/LMSConfigs/BlackboardConfig.jsx

```diff
@@ -44,19 +87,93 @@ const BlackboardConfig = ({ enterpriseCustomerUuid, onClick, existingData }) =>
     }
   };
 
+  const handleAuthorization = async (event) => {
+    event.preventDefault();
+    const transformedConfig = snakeCaseDict(config);
+
+    transformedConfig.active = false;
+    transformedConfig.enterprise_customer = enterpriseCustomerUuid;
+    let err;
+    let configUuid;
+    let fetchedConfigId;
+    // First either submit the new config or update the existing one before attempting to authorize
+    // If the config exists but has been edited, update it
+    if (!isEmpty(existingData) && edited) {
+      try {
+        const response = await LmsApiService.updateBlackboardConfig(transformedConfig, existingData.id);
+        configUuid = response.data.uuid;
+        fetchedConfigId = response.data.id;
+      } catch (error) {
+        err = handleErrors(error);
+      }
+    // If the config didn't previously exist, create it
+    } else if (isEmpty(existingData)) {
+      try {
+        const response = await LmsApiService.postNewBlackboardConfig(transformedConfig);
+        configUuid = response.data.uuid;
+        fetchedConfigId = response.data.id;
+      } catch (error) {
+        err = handleErrors(error);
+      }
+    // else we can retrieve the unedited, existing form's UUID and ID
+    } else {
+      configUuid = existingData.uuid;
+      fetchedConfigId = existingData.id;
+    }
+    if (err) {
+      setErrCode(errCode);
+      openError();
+    } else {
+      // Either collect app key from the existing config data if it exists, otherwise
+      // fetch it from the global config
+      let appKey = existingData.clientId;
+      if (!appKey) {
+        try {
+          const response = await LmsApiService.fetchBlackboardGlobalConfig();
+          appKey = response.data.results[0].app_key;
+        } catch (error) {
+          err = handleErrors(error);
+        }
+      }
+      if (err) {
+        setErrCode(errCode);
+        openError();
+      } else {
+        // Save the config ID so we know one was created in the authorization flow
+        setConfigId(fetchedConfigId);
+        // Reset config polling timeout flag
+        setOauthTimeout(false);
+        // Start the config polling
+        setOauthPollingInterval(LMS_CONFIG_OAUTH_POLLING_INTERVAL);
+        // Start the polling timeout timer
+        setOauthPollingTimeout(LMS_CONFIG_OAUTH_POLLING_TIMEOUT);
+        // Open the oauth window for the user
+        const oauthUrl = `${blackboardBaseUrl}/learn/api/public/v1/oauth2/authorizationcode?`
```

Comment:
not for today would be great toi get this url from backend

Reply:
agreed, probably in a follow up?